File: CS/0804_Unique_Morse_Code_Words.cs
public class Solution {
    public int UniqueMorseRepresentations(string[] words) {
        string[] t = {".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."};
        
        //var set = new HashSet<string>();
        HashSet<string> set = new();
        foreach(string word in words){
            string temp = "";
            foreach(char c in word){
                temp += t[c - 'a'];
            }
            set.Add(temp);
        }
        
        return set.Count;
    }
}

File: Cpp/0804_Unique_Morse_Code_Words.cpp
class Solution {
public:
    int uniqueMorseRepresentations(vector<string>& words) {
        vector<string> t = {".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."};
        int result = 0;
        set<string> s;
        for(string word : words){
            string temp = "";
            for(int i = 0; i < word.size(); i++){
                temp += t[word[i] - 'a'];
            }
            s.insert(temp);
        }
        return s.size();
    }
};
